tests/test_fab_utils.py

```diff
@@ -0,0 +1,17 @@
+from datetime import datetime
+from nose.tools import assert_equal
+
+from commcare_cloud.fab.utils import get_changelogs_in_date_range
+
+
+def test_get_changelogs():
+    assert_equal(
+        set(get_changelogs_in_date_range(datetime(2022, 11, 1), datetime(2022, 11, 11))),
```

Comment:
Since this relies on fetching actual data I suspect it may be flaky - it might be better to mock it to return the content from the local repo itself instead of from the web.

Reply:
I have modified it as as suggested.
 
But, while the current downside is the test duration, the downside if we mock it is that if the Github API or its Python client has a bug the tests will no longer catch that issue.

I think this is a general problem as well with any code that needs to get data from external APIs. May be the ideal setup is to have a separate external API test matrix which we can test on a sparse weekly schedule or such?

Curious to hear your take on this problem.

Reply:
> I think this is a general problem as well with any code that needs to get data from external APIs. May be the ideal setup is to have a separate external API test matrix which we can test on a sparse weekly schedule or such?

For a critical function I would say yes but in this case a breaking change will get is not critical.

On that note, do you think we should catch any exceptions here to prevent blocking the deploy if it fails?

Reply:
Since the intention is to allow the deploy only after changelogs is reviewed, I think blocking it on exception is fine here. The tests ensures an exception free deploy process, but if there is an exception it will be good to hear about for which a hard fail is necessary. Otherwise, this feature might never work and we will never know.

Reply:
We could still show a message to indicate that there was an error and direct the user to check the changelog manually.

>  Otherwise, if this feature breaks we will never know.

Since this will impact our own deploys as well I think we'll notice if there is an error message.

Also, this is just for convenience so I really don't think it should block deploy. We don't block deploy if we aren't able to generate the deploy diff either.

Reply:
Fair point, I have added that.